src/hardware.md

```diff
@@ -37,3 +63,271 @@ We can write import scripts to handle virtually any "open" file format that cont
+  different from that of RMF globally. In order to transform a pose from the RMF
+  frame to the free fleet robot frame, it is first scaled, rotated, then 
+  translated using these parameters -->
+  <param name="scale" value="0.928"/>
+  <param name="rotation" value="-0.013"/>
+  <param name="translation_x" value="-4.117"/>
+  <param name="translation_y" value="27.26"/>
+
+</node>
+```
+
+Furthermore, an example of this configuration can be found in the repository as well, under the packages `ff_examples_ros1` and `ff_exmaples_ros2`. This example launches the example simulation from `ROBOTIS`, shown [here](https://emanual.robotis.com/docs/en/platform/turtlebot3/simulation/#ros-1-simulation), which has a small simulated world with 3 Turtlebot3 mobile robots, each running its own ROS1 navigation stack.
+
+After successful builds for both ROS1 and ROS2 workspaces, the simulation can be launched following [these instructions](https://github.com/osrf/free_fleet#turtlebot3-simulation), which also includes a ROS2 `free_fleet` server, publishing fleet state messages and accepting mode and navigation requests over ROS2 messages and topics.
+
+#### ROS2 Navigation Stack
+
+Similar to a ROS1 navigation stack, the only difference would be the middleware used onboard the mobile robot, which in this scenario is ROS2. Unfortunately, the ROS2 `free_fleet` client is still currently under development, and this section will be updated once the refactoring, implementation and testing has been done.
+
+The same ready `free_fleet` server implementation in the repository is good to go to work in this scenario, as the interfaces provided by the fleet adapters are still the same ROS2 messages and topics.
+
+In the meantime, if required, users can also implement their own `free_fleet` client, by working with the `free_fleet` library that contains the base implementation and API for the DDS communication. This can be further elaborated in the next section [Custom Navigation Stack](#custom-navigation-stack).
+
+#### Custom Navigation Stack
+
+In this configuration, it is assumed that the software running on the mobile robot was written by the users themselves, therefore users have a good understanding of their own APIs and intefaces. This would be extremely useful, for implementing their own `free_fleet` client wrapper. The block diagram below tries to illustrate this configuration. 
+
+<img src="images/free_fleet_custom_config.png">
+
+Once the custom `free_fleet` client is fully functional, it will be a simple task of launching the same ROS2 `free_fleet` server as mentioned the past section, to work with the fleet adapters through ROS2 messages and topics.
+
+#### Vendor Navigation Stack
+
+This configuration assumes that the provided mobile robot is operating on vendor software and hardware, which in most cases, require more effort when it comes to systems integration. There are 2 ways to go about integrating vendor mobile robots with RMF.
```

Comment:
I'm not clear what the difference is between "Vendor Navigation Stack" and "Custom Navigation Stack". Can you help define (or contrast) the two more explicitly and then I can further review this section?

Reply:
Custom Navigation Stacks will be the ones written by the integrators themselves. They could be a robot vendor that wrote their own software, and is trying to integrate with RMF, which means they can easily modify their own code base and API to work with `free_fleet`.

Vendor Navigation Stacks will be from the perspective of a systems integrator, who bought a robot from a robot vendor, with the software on the robot written by the robot vendor, which the SI is unable to change and only has access to the API provided. A good example will be our partners working with the MiR100 robot, they can't change the code base or API, but in the event that they want a fleet of mix-and-match robots, they can probably go for the approach listed here, but its not recommended, as it might be better to just have a fleet of MiR100 even if its 1 robot. 

I sort of followed the sections mentioned by @codebot on the https://github.com/osrf/ros2multirobotbook/issues/6. It might not be super informative in hindsight, let me know if it is redundant.

Reply:
awesome, thanks for the clarification. I've then modified the two sections and re-named them to "Developer Navigation Stack" and "System Integrator via API". The last one I believe they do not have any access to the Nav Stack so the title is hopefully a bit more directly relevant. Open to your comments/changes for sure.